src/main/java/tech/toparvion/analog/util/config/ChoicesCustomConfigurationLoader.java

```diff
@@ -0,0 +1,80 @@
+package tech.toparvion.analog.util.config;
+
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.boot.env.PropertySourceLoader;
+import org.springframework.boot.env.YamlPropertySourceLoader;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+/**
+ * Environment post processor capable of loading choices list from a custom file.<br/>
+ *
+ * @author Polyudov
+ * @implNote <ul>
+ * <li>There are no logs because the logging subsystem is not initialized yet</li>
```

Comment:
This can become not true providing that the comment below will be accepted (and if).

Reply:
In my opinion ``System.err.println(...)`` cannot be named as "logs".
I suggest adding some words to this comment. Something like that:
> There are no logs *(by slf4j loggers)* because the logging subsystem is not initialized yet

I hope this small improvement makes that comment clearer. Ok?

Reply:
Yes, that's better.